source/extensions/filters/listener/proxy_protocol/proxy_protocol.h

```diff
@@ -23,19 +23,54 @@ namespace ProxyProtocol {
 using KeyValuePair =
     envoy::extensions::filters::listener::proxy_protocol::v3::ProxyProtocol::KeyValuePair;
 
+enum ProxyProtocolVersion { Unknown = 0, V1 = 1, V2 = 2 };
```

Comment:
```suggestion
enum ProxyProtocolVersion { NotFound = 0, V1 = 1, V2 = 2 };
```

I think some of this would be cleaner if this only had v1 and v2 in it; treating the error case as a version leads to some weirdness, eg having a stat for errors on the unknown version that can't be incremented.